scripts/thaliPrerequisite.js

```diff
@@ -0,0 +1,112 @@
+var dummyFile = rootdir + '/../plugins/org.thaliproject.p2p/dummy.js';
+
+//Replace JXcoreExtension.java
+//Copy from \plugins\org.thaliproject.p2p\src\android\java\io\jxcore\node to \platforms\android\src\io\jxcore\node
+function replaceJXCoreExtension(){
+    var sourceFile =  rootdir + '/../plugins/org.thaliproject.p2p/src/android/java/io/jxcore/node/JXcoreExtension.java';
+    var targetFile = rootdir + '/../platforms/android/src/io/jxcore/node/JXcoreExtension.java';
+    fs.writeFileSync(targetFile, fs.readFileSync(sourceFile));
+}
+
+//Check whether this is the first time this script is getting executed or not
+//When ever this script run, it creates the dummyfile at the end of execution
+function isFirstTime(){
+    if (fs.existsSync(dummyFile))
+        return false;
+    else
+        return true;
+}
+
+
+//Install thali node modules to global
+//once we have the thali module available in npm repo, this is step is not required
+function installThaliModules(){
+    var thaliPath = rootdir + '/../plugins/org.thaliproject.p2p/thali';
+    var currentFolder = process.cwd();
+    process.chdir(thaliPath);
+    //exec('jx install -g --autoremove "*.gz"'); // https://github.com/jxcore/jxcore/issues/429
+    exec('jx install -g');
+    process.chdir(currentFolder);
+}
+
+
+//TODO: This is a temporary fix, and we don't require this method once we get the JXCore fix in master
+//Uncomment the dependency section in plugin.xml while removing this method
+function updateJXCore(){
+    process.chdir(process.cwd());
+    console.log('Cloning the jxcore-cordova repo (380MB size)..it might take a while. Please be patient..');
+    exec('git clone -b 0.0.3-dev --single-branch https://github.com/jxcore/jxcore-cordova.git');
+    // Another temporary fix to prevent this script is getting called during the following 'plugin add'
+    fs.writeFileSync(dummyFile, '', 'utf8');
+    exec('cordova plugin add jxcore-cordova');
+}
+
+//TODO: This is temporary fix, and we don't reqiure it once the version update PR available
+//Update the Android SDK version
+// ..\platforms\android and in AndroidManifest.xml change android:minSdkVersion="10" to android:minSdkVersion="16"
+function updateAndroidSDKVersion() {
+    var to_replace = 'android:minSdkVersion=\"10\"';
+    var replace_with = 'android:minSdkVersion=\"16\"';
+    var filename = rootdir + '/../platforms/android/AndroidManifest.xml';
+    var data = fs.readFileSync(filename, 'utf8');
+    var result = data.replace(new RegExp(to_replace, "g"), replace_with);
+    fs.writeFileSync(filename, result, 'utf8');
+}
+
+//Copy the build-extras.gradle
+//Copy from \plugins\org.thaliproject.p2p\build to \platforms\android
+function copyBuildExtras(){
+    var sourceFile =  rootdir + '/../plugins/org.thaliproject.p2p/build/build-extras.gradle';
+    var targetFile = rootdir + '/../platforms/android/build-extras.gradle';
+    fs.writeFileSync(targetFile, fs.readFileSync(sourceFile));
+}
+
+
+if(isFirstTime())
```

Comment:
It's generally considered best practice to keep code as flat as possible. So in a case like this we would want the code to probably look something like:

```Node
if (isFirstTime() == false) {
  return;
}
console.log('Starting the....
```

I should point out however that this isn't strictly idiomatic. The idiomatic approach would use `!isFirstTime()` but as I mentioned previously I hate !s because I think they are easy to miss when reading code. But that is my own personal hangup.